### PR TITLE
[Labs] Make qudit MMD loss modular

### PR DESCRIPTION
**Pull request by willjmax**

### CODE REVIEW

File: pennylane/labs/tcdq/qudit_mmd_loss.py
Comment (josephbowles) [inline pennylane/labs/tcdq/qudit_mmd_loss.py:64]: would be good to have the bootstrap option like in the qubit PR as well. I can do a quick PR once this is merged.

File: pennylane/labs/tcdq/qudit_mmd_loss.py [hunk 1]
```diff
@@ -271,93 +273,123 @@ def _unbiased_mmd_squared(  # pylint: disable=too-many-arguments
...
     Args:
-        circuit_config (QuditCircuitConfig): Qudit circuit description
-            specifying gate structure, qudit dimension, and sample
-            count. See :class:`~pennylane.labs.tcdq.QuditCircuitConfig`.
+        expval_fn (Callable): Heisenberg-Weyl expectation value function, as
+            above. Must be hashable and JAX-traceable.
+        dims (int | Sequence[int]): Local qudit dimension(s). Either a single
+            ``int`` broadcast to every qudit, or a sequence of length
+            ``n_qudits`` giving a distinct dimension :math:`d_j` per qudit.
+        n_qudits (int): Number of qudits the model acts on, i.e. the width of
```
Comment (josephbowles) [inline pennylane/labs/tcdq/qudit_mmd_loss.py:368]: isn't this implied directly from `len(dims)`?

**Review by josephbowles — APPROVED** [review]

File: pennylane/labs/tcdq/qudit_mmd_loss.py [hunk 2]
```diff
@@ -474,16 +509,14 @@ def loss_fn(
                 eval_key=eval_key,
                 params=jnp.asarray(params),
```
Comment (josephbowles) [inline pennylane/labs/tcdq/qudit_mmd_loss.py]:
```suggestion
                params=params,
```

File: pennylane/labs/tcdq/qudit_mmd_loss.py
Comment (willjmax) [reply pennylane/labs/tcdq/qudit_mmd_loss.py:64]: Yes, let's do that in a separate PR.

**Review by austingmhuang — APPROVED** [review]

File: pennylane/labs/tcdq/__init__.py [hunk 3]
```diff
@@ -300,6 +307,11 @@ def loss_fn(current_params):
 
    print("Final MMD loss:", float(result.losses[-1]))
 
+Any extra keyword arguments needed by the moment callable are forwarded
+through the loss, for example
+``loss_fn(params, target_data, key, n_samples=8000)`` or
+``loss_fn(params, target_data, key, phase_fn_params=xi)``.
```
Comment (austingmhuang) [inline pennylane/labs/tcdq/__init__.py:313]: Can be made clearer I think. Like what exactly is forwarded? Just `n_samples` is forwarded to the moment callable?
